test/chromium-test.js

```diff
@@ -121,6 +121,32 @@ describe('chromium', function () {
     });
   });
 
+  it('enables dumpio non-silent]', function (done) {
+    run('fails', ['-R', 'dot', '--no-colors', '--dumpio'],
+      function (code, stdout) {
+        var expected = '# chromium:\n\n\n'
+        + 'Error: Oh noes!\n'
+        + '      at test/fails.js:12\n'
+        + '  !!..\n'
+        + '\n'
+        + '  2 passing (115ms)\n'
+        + '  2 failing\n'
+        + '\n'
+        + '  1) test\n'
+        + '       fails synchronously:\n'
+        + '     Error: Oh noes!\n'
+        + '      at Context.<anonymous> (test/fails.js:6)\n'
+        + '\n'
+        + '  2) test\n'
+        + '       fails asynchronously:\n'
+        + '     Uncaught Error: Script error. (:0)';
+
+        assert.equal(stdout.slice(0, 40), expected.slice(0, 40));
```

Comment:
I don't understand this assertion. The `stdout` value does not seem to be relevant for this test.

Reply:
Pardon for being unclear - I could have explained this in a comment of some sort:

Basically, there is no output if `--dumpio` is not given. A failing test just results in `''` - the empty string. So this is basically just a way of asserting that the output is produced at all.

If I remove the `--dumpio` option the test fails with this:

```javascript
      Uncaught AssertionError [ERR_ASSERTION]: '' == '# chromium:\n\n\nError: Oh noes!\n      at t'
      + expected - actual

      +# chromium:
      +
      +
      +Error: Oh noes!
      +      at t
```

Reply:
I couldn't come up with any other way of asserting that this worked, but I take suggestions :smiley_cat:

Reply:
Hmmm, I guess I have to play with this myself to understand. Seems dodgy that regular test output is swallowed if this flag is not given. It should always do that, IMO.

Reply:
So, I check out you code and found that it makes no difference if I'm removing `--dumpio`, the test is still green. The flag causes puppeteer to write a lot of detailed information to `stderr`, which is passed back to the test as a third argument. I guess that the test should assert.

I also ran tests on other projects with this flag and since it dumps _all_ IO, the test output is shown twice. So it would make sense not print anything to standard out ourselves if `--dumpio` is given. Makes sense?

Reply:
Oh, man. I have no idea how that happened. I was *so sure* I had tested with/without the dumpio option and saw different behaviour. It must be because I worked from morning until afternoon without eating. That obviously just earned me an award in the league of morons  :1st_place_medal:  

Fixing it now, but I am still not sure how to do a proper part-wise match on the output. Unless you get a better suggestion, I'll stick to slicing on sub-parts of the output, which feels less than great ...

So for the following output, I'd match on `DevTools listening on ws://127.0.0.1`, `[.*:INFO:CONSOLE(.*)]`, `source: __puppeteer_evaluation_script__` and the output from stdout.
```
DevTools listening on ws://127.0.0.1:37763/devtools/browser/90ab4733-514e-445b-b88a-97915efbe68d
[0322/153940.105525:ERROR:command_buffer_proxy_impl.cc(106)] ContextResult::kTransientFailure: Shared memory region is not valid
[0322/153940.333397:INFO:CONSOLE(1415)] "", source: __puppeteer_evaluation_script__ (1415)
[0322/153940.337272:INFO:CONSOLE(1415)] "", source: __puppeteer_evaluation_script__ (1415)
[0322/153940.391361:INFO:CONSOLE(1347)] "Uncaught Error: Oh noes!", source: __puppeteer_evaluation_script__ (1347)
[0322/153940.450332:INFO:CONSOLE(1415)] "  !!..", source: __puppeteer_evaluation_script__ (1415)
[0322/153940.450540:INFO:CONSOLE(1415)] "", source: __puppeteer_evaluation_script__ (1415)
[0322/153940.450926:INFO:CONSOLE(1415)] "  2 passing (119ms)", source: __puppeteer_evaluation_script__ (1415)
[0322/153940.451132:INFO:CONSOLE(1415)] "  2 failing", source: __puppeteer_evaluation_script__ (1415)
[0322/153940.451323:INFO:CONSOLE(1415)] "", source: __puppeteer_evaluation_script__ (1415)
[0322/153940.451818:INFO:CONSOLE(1415)] "  1) test", source: __puppeteer_evaluation_script__ (1415)
[0322/153940.452002:INFO:CONSOLE(1415)] "       fails synchronously:", source: __puppeteer_evaluation_script__ (1415)
[0322/153940.452168:INFO:CONSOLE(1415)] "     Error: Oh noes!", source: __puppeteer_evaluation_script__ (1415)
[0322/153940.452339:INFO:CONSOLE(1415)] "      at Context.<anonymous> (__puppeteer_evaluation_script__:1341:11)", source: __puppeteer_evaluation_script__ (1415)
[0322/153940.452476:INFO:CONSOLE(1415)] "      at callFn (__puppeteer_evaluation_script__:6032:21)", source: __puppeteer_evaluation_script__ (1415)
[0322/153940.452617:INFO:CONSOLE(1415)] "      at Test.Runnable.run (__puppeteer_evaluation_script__:6024:7)", source: __puppeteer_evaluation_script__ (1415)
[0322/153940.452857:INFO:CONSOLE(1415)] "      at Runner.runTest (__puppeteer_evaluation_script__:6560:10)", source: __puppeteer_evaluation_script__ (1415)
[0322/153940.452942:INFO:CONSOLE(1415)] "      at __puppeteer_evaluation_script__:6678:12", source: __puppeteer_evaluation_script__ (1415)
[0322/153940.453063:INFO:CONSOLE(1415)] "      at next (__puppeteer_evaluation_script__:6474:14)", source: __puppeteer_evaluation_script__ (1415)
[0322/153940.453217:INFO:CONSOLE(1415)] "      at __puppeteer_evaluation_script__:6484:7", source: __puppeteer_evaluation_script__ (1415)
[0322/153940.453382:INFO:CONSOLE(1415)] "      at next (__puppeteer_evaluation_script__:6408:14)", source: __puppeteer_evaluation_script__ (1415)
[0322/153940.453476:INFO:CONSOLE(1415)] "      at __puppeteer_evaluation_script__:6452:5", source: __puppeteer_evaluation_script__ (1415)
[0322/153940.453701:INFO:CONSOLE(1415)] "      at timeslice (__puppeteer_evaluation_script__:1564:27)", source: __puppeteer_evaluation_script__ (1415)
[0322/153940.453781:INFO:CONSOLE(1415)] "", source: __puppeteer_evaluation_script__ (1415)
[0322/153940.453872:INFO:CONSOLE(1415)] "  2) test", source: __puppeteer_evaluation_script__ (1415)
[0322/153940.454070:INFO:CONSOLE(1415)] "       fails asynchronously:", source: __puppeteer_evaluation_script__ (1415)
[0322/153940.454141:INFO:CONSOLE(1415)] "     Uncaught Error: Script error. (:0)", source: __puppeteer_evaluation_script__ (1415)
[0322/153940.454292:INFO:CONSOLE(1415)] "      at global.onerror (__puppeteer_evaluation_script__:1546:10)", source: __puppeteer_evaluation_script__ (1415)
[0322/153940.454414:INFO:CONSOLE(1415)] "", source: __puppeteer_evaluation_script__ (1415)
[0322/153940.454499:INFO:CONSOLE(1415)] "", source: __puppeteer_evaluation_script__ (1415)
[0322/153940.454667:INFO:CONSOLE(1415)] "", source: __puppeteer_evaluation_script__ (1415)
[0322/153940.455152:INFO:CONSOLE(12)] "[EXIT 1]", source: __puppeteer_evaluation_script__ (12)
```

Reply:
Don't be so hard on yourself 😉 I guess it doesn't even have to be the entire output. It should be good enough to just find a line or some part of the output that is unique, like, checking if `__puppeterr_evaluation_source` exists in the string.

Reply:
It's fixed. Check again.